presto-client/src/main/java/io/prestosql/client/ClientTypeSignature.java

```diff
@@ -108,6 +115,9 @@ private String rowToString()
                 })
                 .collect(joining(","));
 
+        if (isNullOrEmpty(fields)) {
```

Comment:
`fields` cannot be null, so just:

```suggestion
        if (fields.isEmpty()) {
```